hudi-cli/src/main/java/org/apache/hudi/cli/commands/ClusteringCommand.java

```diff
@@ -0,0 +1,102 @@
+    String sparkPropertiesPath =
+        Utils.getDefaultPropertiesFile(JavaConverters.mapAsScalaMapConverter(System.getenv()).asScala());
+    SparkLauncher sparkLauncher = SparkUtil.initLauncher(sparkPropertiesPath);
+
+    // First get a clustering instant time and pass it to spark launcher for scheduling clustering
+    String clusteringInstantTime = HoodieActiveTimeline.createNewInstantTime();
+
+    sparkLauncher.addAppArgs(SparkCommand.CLUSTERING_SCHEDULE.toString(), client.getBasePath(),
+        client.getTableConfig().getTableName(), clusteringInstantTime, sparkMemory, propsFilePath);
+    UtilHelpers.validateAndAddProperties(configs, sparkLauncher);
+    Process process = sparkLauncher.launch();
+    InputStreamConsumer.captureOutput(process);
+    int exitCode = process.waitFor();
+    if (exitCode != 0) {
+      return "Failed to schedule clustering for " + clusteringInstantTime;
+    }
+    return "Attempted to schedule clustering for " + clusteringInstantTime;
+  }
+
+  @CliCommand(value = "clustering run", help = "Run Clustering")
+  public String runClustering(
+      @CliOption(key = "parallelism", help = "Parallelism for hoodie clustering",
+          unspecifiedDefaultValue = "1") final String parallelism,
+      @CliOption(key = "sparkMemory", help = "Spark executor memory",
+          unspecifiedDefaultValue = "4G") final String sparkMemory,
+      @CliOption(key = "retry", help = "Number of retries",
+          unspecifiedDefaultValue = "1") final String retry,
+      @CliOption(key = "clusteringInstant", help = "Clustering instant time",
+          mandatory = true) final String clusteringInstantTime,
+      @CliOption(key = "propsFilePath", help = "path to properties file on localfs or dfs with configurations for hoodie client for compacting",
+          unspecifiedDefaultValue = "") final String propsFilePath,
+      @CliOption(key = "hoodieConfigs", help = "Any configuration that can be set in the properties file can be passed here in the form of an array",
+          unspecifiedDefaultValue = "") final String[] configs
+  ) throws Exception {
+    HoodieTableMetaClient client = HoodieCLI.getTableMetaClient();
+    String sparkPropertiesPath =
+        Utils.getDefaultPropertiesFile(JavaConverters.mapAsScalaMapConverter(System.getenv()).asScala());
+    SparkLauncher sparkLauncher = SparkUtil.initLauncher(sparkPropertiesPath);
+    sparkLauncher.addAppArgs(SparkCommand.CLUSTERING_RUN.toString(), client.getBasePath(),
+        client.getTableConfig().getTableName(), clusteringInstantTime, parallelism, sparkMemory, retry, propsFilePath);
+    UtilHelpers.validateAndAddProperties(configs, sparkLauncher);
+    Process process = sparkLauncher.launch();
+    InputStreamConsumer.captureOutput(process);
+    int exitCode = process.waitFor();
+    if (exitCode != 0) {
+      return "Failed to run clustering for " + clusteringInstantTime;
+    }
+    return "Clustering successfully completed for " + clusteringInstantTime;
```

Comment:
Succeed to run clustering for " + clusteringInstantTime

Reply:
Updated.